### PR TITLE
options.logo support string array

### PR DESCRIPTION
**Pull request by carlhopf**
favicons >= 7.0.0 picks source image by it's size if provided with an array

solves https://github.com/jantimon/favicons-webpack-plugin/issues/265

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -110,20 +114,33 @@ class FaviconsWebpackPlugin {
...
-          ([logo]) =>
-            getRelativeOutputPath(logo.hash, compilation, this.options),
-          ([logo, manifest], getRelativeOutputPath) =>
-            this.generateFavicons(
-              logo,
-              manifest.content,
+          // Recompile filesystem cache if any source based path change:
+          (fileSources) =>
+            getRelativeOutputPath(
+              fileSources[0].hash,
```
Comment (andy128k) [inline src/index.js:120]: Better to hash all files.

File: src/index.js [hunk 2]
```diff
@@ -286,12 +303,17 @@ class FaviconsWebpackPlugin {
    * @param {import('webpack').Compilation} compilation
    * @param {string} outputPath
    */
-  generateFavicons(logo, baseManifest, compilation, outputPath) {
+  generateFavicons(logoFileSources, baseManifest, compilation, outputPath) {
     const resolvedPublicPath = getResolvedPublicPath(
-      logo.hash,
+      logoFileSources[0].hash,
```
Comment (andy128k) [inline src/index.js:308]: Hash all files

**Review by andy128k — CHANGES_REQUESTED** [review]
Great work!

My only concern is that only first file's hash is used for caching. Cache invalidation is known to be [a hard thing](https://martinfowler.com/bliki/TwoHardThings.html) ;),

May you check if this change #334 can be useful here?

File: src/index.js (hunk 1 above)
Comment (andy128k) [reply src/index.js:120]: Smth like `getContentHash(...fileSources.map(s => s.content))` should work.

File: src/index.js (hunk 2 above)
Comment (carlhopf) [reply src/index.js]: amended 👍

**Comment by carlhopf** [comment]
ah took me a bit to wrap my head around the internals ;) thanks for the example 👍  updated commit pushed

**Comment by andy128k** [comment]
Thanks!

**Pull request merged**